src/parser.rs

```diff
@@ -3441,6 +3448,9 @@ impl<'a> Parser<'a> {
                     Some('\'') => Ok(Value::SingleQuotedString(w.value)),
                     _ => self.expected("A value?", Token::Word(w))?,
                 },
+                Keyword::NoKeyword if dialect_of!(self is SnowflakeDialect) => {
```

Comment:
@yuval-illumex adding the generic here is feasible?

I'd suggest adding it but, as identifiers have the same exact syntax, I don't know how that would work.

Reply:
@AugustoFKL added :)